google-cloud-storage/lib/google/cloud/storage/file/signer_v2.rb

```diff
@@ -129,11 +130,15 @@ def signed_url options
           end
 
           def generate_signature signing_key, secret
-            unless signing_key.respond_to? :sign
-              signing_key = OpenSSL::PKey::RSA.new signing_key
+            if signing_key.is_a? Proc
+              Base64.strict_encode64(signing_key.call(secret)).delete "\n"
```

Comment:
Consider

```ruby
signature = signing_key.call secret
```

and then move the `Base64.strict_encode64` call outside the if statement since it's the same as below.